src/commands/server/deploy.ts

```diff
@@ -423,6 +428,37 @@ export default class Deploy extends Command {
   }
 }
 
+function ensureOIDCProviderInstalled(flags: any): Listr.ListrTask {
+  return {
+    title: 'Check if OIDC Provider installed',
+    enabled: () => !flags['skip-oidc-provider-check'] && isKubernetesPlatformFamily(flags.platform),
+    skip: () => {
+      if (flags.platform === 'minikube') {
+        return 'Dex will be automatically installed'
+      }
+    },
+    task: async (_ctx: any, task: any) => {
+      const kube = new KubeHelper(flags)
+      const apiServerPods = await kube.getPodListByLabel('kube-system', 'component=kube-apiserver')
+      for (const pod of apiServerPods) {
+        if (!pod.spec) {
+          continue
+        }
+        for (const container of pod.spec.containers) {
+          if (container.command) {
+            if (container.command.some(value => value.includes('oidc-username'))) {
```

Comment:
https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuring-the-api-server regarding to this, only `oidc-issuer-url` and `oidc-client-id` are required, so I would use those.